src/app/pages/courses/course/course.component.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  Component,
+  OnInit,
+  OnChanges,
+  EventEmitter,
+  SimpleChanges,
+  Input,
+  Output,
+  ChangeDetectionStrategy
+} from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+import { CoursesService } from '../../../services/courses.service';
+import { AppState } from '../../../app.service';
+import { ICourse } from '../iCourse.interface';
+import { CanActivate, Router, ActivatedRoute } from '@angular/router';
+import { NgForm } from '@angular/forms';
+
+@Component({
+  selector: 'app-course',
+  providers: [],
+  styleUrls: ['./course.component.scss'],
+  templateUrl: './course.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CourseComponent implements OnInit {
+  @Input() courseData: ICourse;
+  @Output() onVoted = new EventEmitter<boolean>();
```

Comment:
spoko, też tworze taki kod, żeby coś popatrzeć, ale jak wiem, że jest nieużywany to wtedy go komentuję, żeby komuś kto review'uje nie mieszać w głowie ;)